### PR TITLE
Add .fidl support

### PR DESCRIPTION
**Pull request by chaopeng**
Add [FIDL](https://fuchsia.dev/fuchsia-src/reference/fidl/language/language) (Fuchsia Interface Definition Language) support.

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -535,6 +535,16 @@ list.fennel = {
   maintainers = { "@alexmozaidze" },
 }
 
+list.fidl = {
+  install_info = {
+    url = "path/to/tree-sitter-fidl",
+    files = { "src/parser.c" },
+    requires_generate_from_grammar = true,
+  },
+  filetype = "fidl",
```
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:544]:
```suggestion
```
Not needed if filetype is the same as language name

**Comment by clason** [comment]
Please 

1. add a lockfile.json entry
2. add injection queries at least for comments
3. run `./scripts/format-queries.lua queries/fidl`

File: lua/nvim-treesitter/parsers.lua [hunk 2]
```diff
@@ -535,6 +535,16 @@ list.fennel = {
   maintainers = { "@alexmozaidze" },
 }
 
+list.fidl = {
+  install_info = {
+    url = "path/to/tree-sitter-fidl",
```
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:540]: Nice try. That should be an actual Github URL.

**Comment by chaopeng** [comment]
All comments resolved, ptal thank you

File: queries/fidl/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  "struct"
+  "table"
+  "type"
+  "union"
+  "using"
+] @keyword
+
+(primitives_type) @type
+
+(builtin_complex_type) @type
```
Comment (lucario387) [inline queries/fidl/highlights.scm:30]: should be `@type.builtin`

File: queries/fidl/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  "service"
+  "strict"
+  "struct"
+  "table"
+  "type"
+  "union"
+  "using"
+] @keyword
+
+(primitives_type) @type
```
Comment (lucario387) [inline queries/fidl/highlights.scm:28]: Is this also builtin type?

File: queries/fidl/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  "|"
+  "&"
+  "->"
+] @operator
+
+(attribute
+  (identifier) @attribute)
+
+(attribute
+  "@" @attribute)
```
Comment (lucario387) [inline queries/fidl/highlights.scm:46]: If there's only 1 identifier for an attribute, consider grouping these 2 into one
```suggestion
(attribute
  "@" @attribute
  (identifier) @attribute)
```

File: queries/fidl/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  (identifier) @attribute)
+
+(string_literal) @string
+
+(numeric_literal) @number
+
+[
+  (true)
+  (false)
+] @constant.builtin
```
Comment (lucario387) [inline queries/fidl/highlights.scm:53]: `@boolean`

**Review by lucario387 — APPROVED** [review]

**Pull request merged**